net/sched/sch_dsprio.c

```diff
@@ -269,8 +286,10 @@ static void dsprio_reset(struct Qdisc *sch)
 	sch->qstats.backlog = 0;
 	sch->q.qlen = 0;
 
-	for (prio = 0; prio < DSPRIO_MAX_PRIORITY; prio++)
+	for (prio = 0; prio < DSPRIO_MAX_PRIORITY; prio++) {
 		__skb_queue_purge(&q->qdiscs[prio]);
+		memset(&q->qstats[prio], 0, sizeof(q->qstats[prio]));
```

Comment:
You can put this `memset()` outside of the loop and apply it to the whole array.